### ISSUE TITLE
Wrong type for var

### ISSUE THREAD

**Issue by kjnsn** [dgraph-io/dgraph#857]
The following query is returning "Wrong variable type encountered for var(score) 2."

```{
    var(id: 0xd244867601d87b76) {
        name@en
        ~genre (first: 20) {
        starring {
          ACTORS as performance.actor {
          }
        }
      }
  }

  A as var(id: 0x36692145960cfceb) {}

  var(id: 0xd244867601d87b76) {
    films as ~genre {
      starring {
        score as count(performance.actor @filter(var(ACTORS) or var(A)))
      }
    }
  }
  Recs(id: var(films), orderdesc: var(score), first: 10) {
    name@en
    var(score)
  }
}```

**Comment by ashwin95r** [comment]
Fixed. merged to release 0.7.5

**Comment by ashwin95r** [comment]
Closing. Feel free to reopen if the problem persists.
